src/Component/AutoMapper/Generator/Generator.php

```diff
@@ -161,8 +161,10 @@ public function generate(MapperGeneratorMetadataInterface $mapperGeneratorMetada
 
             $transformer = $propertyMapping->getTransformer();
 
-            $sourcePropertyAccessor = $propertyMapping->getReadAccessor()->getExpression($sourceInput);
-            [$output, $propStatements] = $transformer->transform($sourcePropertyAccessor, $result, $propertyMapping, $uniqueVariableScope);
+            $fieldValueVariable = new Expr\Variable($uniqueVariableScope->getUniqueName('fieldValue'));
+            $sourcePropertyAccessor = new Expr\Assign($fieldValueVariable, $propertyMapping->getReadAccessor()->getExpression($sourceInput));
+
+            [$output, $propStatements] = $transformer->transform($fieldValueVariable, $result, $propertyMapping, $uniqueVariableScope);
```

Comment:
I've introduced a new variable in order no to repeat the whole new code block added in `ReadAccessor`